### PR TITLE
Implement `snapcraft push`

### PR DESCRIPTION
**Pull request by sergiusens**

### CODE REVIEW

File: snapcraft/main.py [hunk 1]
```diff
@@ -79,7 +83,8 @@
   register     Register the package name in the store.
   tour         Setup the snapcraft examples tour in the specified directory,
                or ./snapcraft-tour/.
-  upload       Upload a snap to the Ubuntu Store.
+  push         Pushes and optionally releases a snap to the Ubuntu Store.
+  upload       DEPRECATED Upload a snap to the Ubuntu Store.
```
Comment (josepht) [inline snapcraft/main.py]: this should probably indicate that 'push' replaces it as well.

**Comment by josepht** [comment]
It seems strange to have 'push' accept multiple channels but 'release' only accept one.   Of course that's previous work and not related to this PR.  LGTM

**Comment by sergiusens** [comment]
El 13/07/16 a las 16:30, Joe Talbott escribió:
> It seems strange to have 'push' accept multiple channels but 'release'
> only accept one. Of course that's previous work and not related to this
> PR. LGTM

I am just following the designed UX :-)

**Comment by elopio** [comment]
:+1:, minus the message in upload help that Joe mentioned.
For the future, git mv would have made the diff a lot nicer.

**Comment by sergiusens** [comment]
El 13/07/16 a las 19:40, Leo Arias escribió:
> 👍, minus the message in upload help that Joe mentioned.
> For the future, git mv would have made the diff a lot nicer.

This was done with git mv! ;-)

**Pull request merged**